### PR TITLE
Adding layout analysis python script

### PR DESCRIPTION
**Pull request by cgtz**
This is the start of a script to analyze the distribution
of partitions in partition layout files. This will be used for
testing the rack aware partition manager.  We can
add features to this tool as we see fit.

*Time to review:* 10 min.
*Primary reviewers:* Gopal, Dmitry

### CODE REVIEW

**Comment by sriramsub** [comment]
This has been around for 8 days now

**Comment by vgkholla** [comment]
I think @cgtz was using this script to evaluate the rack aware partition manager and was waiting for everything to be in a mergeable state (he has made commits a day ago).

File: ambry-tools/scripts/layout-analyzer.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+                partition.add_node(node)
+            for dc, nodes in partition.nodes_by_datacenter.items():
+                self.dc_node_combo_map[dc][frozenset(nodes)].add(partition)
+            self.partition_map[partition_struct["id"]] = partition
+
+    def rack_id(self, node_host, node_port):
+        k = (node_host, node_port)
+        if k in self.node_map:
+            return self.node_map[k].rack_id
+        raise Exception("Node not found")
```
Comment (vgkholla) [inline ambry-tools/scripts/layout-analyzer.py]: minor: add `node_host` and `node_port` to the exception message.

File: ambry-tools/scripts/layout-analyzer.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    def print_report(self):
+        for dc, node_combo_map in self.dc_node_combo_map.items():
+            print("In datacenter: {}".format(dc))
+            max_combo = max(node_combo_map,
+                            key=lambda k: len(node_combo_map[k]))
+            print("Num node combos used: {}".format(len(node_combo_map)))
+            print("Average partitions sharing a node combo: {}".format(
+                sum(len(partitions) for partitions in node_combo_map.values()) / float(len(node_combo_map))))
+            print("Max partitions sharing a node combo: {} on the following nodes:".format(
+                len(node_combo_map[max_combo])))
```
Comment (vgkholla) [inline ambry-tools/scripts/layout-analyzer.py]: you can define a threshold and say that the assignment is bad if max >>> avg

File: ambry-tools/scripts/layout-analyzer.py [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+            for node in max_combo:
+                print(node)
+
+            sum_racks, n_partitions, min_racks = 0, 0, sys.maxsize
+            for partition in self.partition_map.values():
+                num_racks = len(partition.racks_used(dc))
+                if num_racks > 0:
```
Comment (vgkholla) [inline ambry-tools/scripts/layout-analyzer.py]: perhaps we should throw here since this might mean there are no nodes for a partition?

**Comment by vgkholla** [comment]
Minor comments. LGTM.

**Comment by cgtz** [comment]
@vgkholla Addressed comments

**Comment by vgkholla** [comment]
Looks good. Is Dmitry going to take a look? Will merge otherwise.

**Comment by cgtz** [comment]
I think you can go ahead and merge this.

**Pull request merged**